rpcs3/Emu/RSX/GL/GLTexture.cpp

```diff
@@ -319,6 +319,14 @@ namespace rsx
 				std::vector<gsl::byte> staging_buffer)
 			{
 				int mip_level = 0;
+				if (is_compressed_format(format))
+				{
+					//Compressed formats have a 4-byte alignment
+					//TODO: Verify that samplers are not affected by the padding
+					width = align(width, 4);
+					height = align(height, 4);
```

Comment:
It seems some endianness swap is happening during argument fetching as we have discussed with nekotekina on discord. libgcm seems to swap endianness by itself for some methods which may have confused me a bit; will undo the swapping as it seems to be unnecessary.
Confirm that it works with just shifting and masking (probably order is inverted though)
As for vulkan, I really have not touched texture setup there.